app/src/main/java/org/simple/clinic/newentry/form/InputField.kt

```diff
@@ -1,3 +1,5 @@
 package org.simple.clinic.newentry.form
 
-abstract class InputField
+abstract class InputField<T> {
```

Comment:
Should this be an interface instead?

Reply:
I don't know that yet,
1. I might pull out all the validation logic outside the classes and use just composition instead of inheritance.
2. There are special cases around validation across multiple fields and country-specific fields.

The way I am approaching this problem right now is to build it for the current form and then scale it across other countries. I do have a high-level design of how this is going to look like, but we have to wait a see in a couple of PRs.